zrml/swaps/src/lib.rs

```diff
@@ -1455,10 +1462,14 @@ mod pallet {
                         .saturated_into(),
                 )?
                 .saturated_into();
-                let fee_pct = T::RikiddoSigmoidFeeMarketEma::fee(*pool_id)?.saturated_into();
-                let fee_plus_one = BASE.saturating_add(fee_pct);
-                let price_with_fee: u128 = bmul(fee_plus_one, price_without_fee)?;
-                Ok(price_with_fee.saturated_into())
+                if with_fees {
+                    let fee_pct = T::RikiddoSigmoidFeeMarketEma::fee(*pool_id)?.saturated_into();
+                    let fee_plus_one = BASE.saturating_add(fee_pct);
+                    let price_with_fee: u128 = bmul(fee_plus_one, price_without_fee)?;
+                    Ok(price_with_fee.saturated_into())
+                } else {
+                    Ok(price_without_fee.saturated_into())
+                }
```

Comment:
May I ask you, why you still regard with_fees here for Rikiddo?

Reply:
Just missed that, removed on local machine, I will push after sometime along with formatting issue fix.